packages/core/src/animation/Animator.ts

```
@@ -1411,13 +1419,24 @@ export class Animator extends Component {
 
     const animatorLayerData = this._getAnimatorLayerData(layerIndex);
+
+    // Guard against crossFade-to-self/current-dest alias.
+    // statePlayDataMap holds one PlayData per AnimatorState; if the requested cross
+    // target is already on src or dest, getOrCreatePlayData would return the same
+    // instance and resetForPlay would corrupt the active runtime track. Treat as
+    // no-op until lifecycle splits persistent override from transient src/dest tracks.
+    if (
+      animatorLayerData.srcPlayData?.state === crossState ||
+      animatorLayerData.destPlayData?.state === crossState
+    ) {
```

ESLint

[error] Replace `⏎······animatorLayerData.srcPlayData?.state·===·crossState·||⏎······animatorLayerData.destPlayData?.state·===·crossState⏎····` with `animatorLayerData.srcPlayData?.state·===·crossState·||·animatorLayerData.destPlayData?.state·===·crossState`

(prettier/prettier)